### PR TITLE
Extend eslint-rules

### LINKED ISSUE #985: Extend eslint-rules
In Studio we already use eslint-config with few rules.
To keep the code readable, nice, and consistent, I would like to add a few more rules.
I looked at the [eslint rules](https://eslint.org/docs/latest/rules/) and picked out a few that I think are useful.

_Feedback and further suggestions on the rules are of course welcome._

I would like to implement these rules:

- `"arrow-spacing": "error"` 
	-> Enforce spacing before and after the arrow in arrow functions
- `"brace-style": "error"` 
	-> Enforce consistent brace style for blocks
- `"comma-spacing": ["error", { "before": false, "after": true }]` 
	-> Enforces consistent spacing after commas in variable declarations, array literals, object literals, function parameters, and sequences
- `"eol-last": ["error", "always"]` 
	-> Enforces that files end with a newline (LF)
- `"func-call-spacing": ["error", "never"]` 
	-> Disallows spaces between the function name and the opening parenthesis that calls it
- `"indent": ["error", 2]` 
	-> Enforce consistent indentation
- `"linebreak-style": ["error", "unix"]` 
	-> Enforce consistent linebreak style
- `"max-len": ["error", { "code": 120, "ignoreUrls": true }]` 
	-> Enforce a maximum line length
- `"no-multi-spaces": ["warn", { "ignoreEOLComments": true }]` 
	-> Disallow multiple whitespaces around logical expressions, conditional expressions, declarations, array elements, object properties, sequences and function parameters
- `"no-trailing-spaces": "error"` 
	-> Disallow trailing whitespace (spaces, tabs, and other Unicode whitespace characters) at the end of lines
- `"object-curly-spacing": ["error", "always"]` 
	-> Enforce consistent spacing inside braces of object literals, destructuring assignments, and import/export specifiers
- `"semi": ["error", "always"]` 
	-> Require semicolons at the end of statements
- `"space-before-function-paren": ["error", "never"]` 
	-> Disallow spacing before function parentheses
- `"space-infix-ops": "error"` 
	-> Require spacing around operators

I'm not quite sure if these rules are too much:

- `"array-bracket-spacing": ["error", "always"]` 
	-> Enforces consistent spacing inside array brackets
- `"arrow-parens": ["error", "as-needed"]` 
	-> Enforces parentheses around arrow function parameters as-needed
- `"jsx-quotes": ["error", "prefer-double"]` or `"jsx-quotes": ["error", "prefer-single"]` 
	-> Enforce the consistent use of either double or single quotes in JSX attributes

### PR DESCRIPTION
**Pull request by naweidner**
Added more eslint-rules like described [here](https://github.com/elan-ev/opencast-studio/issues/985). 
This resolves #985

Haven't added `"array-bracket-spacing": ["error", "always"]` yet. Every useState declaration will look like this: `const [ state, setState ] = useState()` not quite sure, if I like how it looks. I think I will wait for feedback for this rule.

### CODE REVIEW

**Comment by naweidner** [comment]
Should I add comments to each rule that briefly explain what the rules are for? (like in the linked issue)

**Comment by lkiesow** [comment]
> Should I add comments to each rule that briefly explain what the rules are for? (like in the linked issue)

Since ESLint has a good documentation, and it's really easy to look up the rules (if you don't already understand them from their name alone), I don't think it is necessary.

**Review by lkiesow — APPROVED** [review]

File: .eslintrc.json [hunk 1]
```diff
@@ -8,7 +8,25 @@
...
+    "arrow-spacing": "error",
+    "block-spacing": ["error", "always"],
+    "brace-style": "error",
+    "comma-spacing": ["error", { "before": false, "after": true }],
+    "eol-last": ["error", "always"],
+    "func-call-spacing": ["error", "never"],
+    "indent": ["error", 2, { "SwitchCase": 1, "MemberExpression": "off" }],
+    "jsx-quotes": ["error", "prefer-single"],
+    "linebreak-style": ["error", "unix"],
+    "max-len": ["error", { "code": 120, "ignoreUrls": true }],
```
Comment (LukasKalbertodt) [inline .eslintrc.json:23]: Didn't know about `ignoreUrls`, that's cool!

File: .eslintrc.json [hunk 2]
```diff
@@ -8,7 +8,25 @@
...
+
+    "arrow-parens": ["error", "as-needed"],
+    "arrow-spacing": "error",
+    "block-spacing": ["error", "always"],
+    "brace-style": "error",
+    "comma-spacing": ["error", { "before": false, "after": true }],
+    "eol-last": ["error", "always"],
+    "func-call-spacing": ["error", "never"],
+    "indent": ["error", 2, { "SwitchCase": 1, "MemberExpression": "off" }],
+    "jsx-quotes": ["error", "prefer-single"],
```
Comment (LukasKalbertodt) [inline .eslintrc.json:21]: I just checked and Studio used more double quotes before. Enabling this rule causes 77 errors. Enabling `prefer-double` only causes 23 errors. Tobira uses double quotes; the editor seems to use double quotes as well. It's also closer to HTML (which JSX wants to mimic). So I'd have preferred to use double quotes. 
Should we still change this or just keep it as is now?

File: src/settings.js [hunk 3]
```diff
@@ -136,7 +136,7 @@ export class SettingsManager {
         // the value at the end.
         let obj = rawUrlSettings;
         const segments = key.split('.');
-        segments.slice(0, -1).forEach((segment) => {
+        segments.slice(0, -1).forEach( segment => {
```
Comment (LukasKalbertodt) [inline src/settings.js:139]: The space left of `segment` doesn't belong here, right?

File: src/ui/studio/review/index.js [hunk 4]
```diff
@@ -500,24 +501,24 @@ const Preview = forwardRef(function _Preview({ onTimeUpdate, onReady }, ref) {
...
+                  ).length === recordings.length) {
+                    setDurationsCalculated(true);
+                  }
+                  break;
+                default:
+                  // Appease the linter
+                  break;
               }
-            }}
+            }} // eslint-disable-line
```
Comment (LukasKalbertodt) [inline src/ui/studio/review/index.js:519]: Hu, why is that needed?

**Review by LukasKalbertodt — COMMENTED** [review]
Very nice! That makes stuff way easier.

I realize this is already merged, but I thought my comments might be useful anyway.
